src/Parser/Hunter/Survival/Modules/Spells/WildfireBomb.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+
+import SPELLS from 'common/SPELLS';
+import { formatPercentage } from 'common/format';
+import StatisticBox from 'Interface/Others/StatisticBox';
+import SpellIcon from 'common/SpellIcon';
+import Enemies from 'Parser/Core/Modules/Enemies';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+import StatTracker from 'Parser/Core/Modules/StatTracker';
+import SpellLink from 'common/SpellLink';
+import STATISTIC_ORDER from 'Interface/Others/STATISTIC_ORDER';
+
+/**
+ * Hurl a bomb at the target, exploding for (45% of Attack power) Fire
+ * damage in a cone and coating enemies in wildfire, scorching them for (90%
+ * of Attack power) Fire damage over 6 sec.
+ */
+
+const GCD_BUFFER = 500; //People aren't robots, give them a bit of leeway in terms of when they cast WFB to avoid capping on charges
+const MS_BUFFER = 200;
+
+class WildfireBomb extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+    spellUsable: SpellUsable,
+    statTracker: StatTracker,
+  };
+
+  acceptedCastDueToCapping = false;
+  currentGCD = 0;
+  badRefreshes = 0;
+  lastRefresh = 0;
+  casts = 0;
+  targetsHit = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = !this.selectedCombatant.hasTalent(SPELLS.WILDFIRE_INFUSION_TALENT.id); //WFI changes WFB so much that a separate module will be added into talents folder to handle everything from that.
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.WILDFIRE_BOMB.id) {
+      return;
+    }
+    this.casts++;
+    this.currentGCD = (Math.max(1.5 / (1 + this.statTracker.currentHastePercentage), 0.75) * 1000);
+    if (this.spellUsable.cooldownRemaining(SPELLS.WILDFIRE_BOMB.id) < GCD_BUFFER + this.currentGCD || !this.spellUsable.isOnCooldown(SPELLS.WILDFIRE_BOMB.id)) {
+      this.acceptedCastDueToCapping = true;
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.WILDFIRE_BOMB_IMPACT.id) {
+      return;
+    }
+    if (this.casts === 0) {
+      this.casts++;
+      this.spellUsable.beginCooldown(SPELLS.WILDFIRE_BOMB.id, this.owner.fight.start_time);
+    }
+    this.targetsHit++;
+    const enemy = this.enemies.getEntity(event);
+    if (this.acceptedCastDueToCapping || !enemy) {
+      return;
+    }
+    if (enemy.hasBuff(SPELLS.WILDFIRE_BOMB_DOT.id) && event.timestamp > this.lastRefresh + MS_BUFFER) {
```

Comment:
The second part of the condition will basically always return true (unless the bomb can hit the same target multiple times from one cast?). Should it also be considered a bad refresh if you're AOEing and reduce the cooldown below the dot duration?

Reply:
I only wanted to count it once during AoE, so first one will be true, the subsequent will be false until a new cast comes along. 

No AoE'ing and getting it below DoT duration should be fine (it can happen with Butchery), but should be more edge cases - and if you do so I can imagine Serpent Sting modules will start getting cranky because it'll come at the cost of that uptime.